src/openrct2/interface/Screenshot.cpp

```diff
@@ -143,6 +129,12 @@ static int32_t screenshot_get_next_path(char* path, size_t size)
         fileNameCh, leftBytes, "%s %d-%02d-%02d %02d-%02d-%02d.png", park_name, currentDate.year, currentDate.month,
         currentDate.day, currentTime.hour, currentTime.minute, currentTime.second);
 
+    if (!filename_valid_characters(fileNameCh))
```

Comment:
This isn't exactly a fix if you just disallow the saving of screenshots due to the park name, this replaces one error with another. I think it would be better to simply sanitize the filename that will work across all platforms The screenshot file name does not need the exact park name in it as long it can be easily identified still.

Reply:
Please also have a look at https://github.com/OpenRCT2/OpenRCT2/wiki/Commit-Messages

Reply:
The original code sanitised colons which you have now removed. It should instead replace all invalid characters with a valid character like underscore or hyphen. Also note that `filename_valid_characters` is not a platform function so each OS would end up sanitising the same characters regardless of their support.

Reply:
Thanks for your comments. Taking them into account, I am thinking of adding platform functions for determining the valid characters. Maybe analogous to [Microsoft's](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getinvalidpathchars?view=netframework-4.7.2). 

Any resources on the utf8 string classes by the way ?

Reply:
You mean this: https://github.com/OpenRCT2/OpenRCT2/blob/7de115263e32a3b1daf1aa5bbc81479aa58e278d/src/openrct2/common.h#L27-L28 ?

Reply:
@tomlankhorst You should use `std::string_view` where possible.